src/main/webapp/app/exercises/programming/manage/code-editor/code-editor-instructor-base-container.component.ts

```diff
@@ -139,9 +139,9 @@ export abstract class CodeEditorInstructorBaseContainerComponent implements OnIn
                         this.exercise.exerciseHints = exerciseHints;
                         this.loadingState = LOADING_STATE.CLEAR;
                     },
-                    (err) => {
+                    (err: Error) => {
                         this.loadingState = LOADING_STATE.FETCHING_FAILED;
-                        this.onError(err);
+                        this.onError(err.message);
```

Comment:
@julian-christl E.g. in this case the subscription error-block actually expected a string instead of a proper error. Therefore, throwing just a string above was fine.
As there were only a few places where strings instead of errors were used, I replaced them with proper errors to be consistent everywhere.
Sadly there is no type safety between the throws and the catch blocks.

Reply:
I think it would make sense to let `this.onError()` expect an error instead of a message, don't you think?
Not sure if this is totally out of scope.

Reply:
In this case the message is actually an error identifier that is added to another prefix to get the correct translation string `artemisApp.editor.errors.${error}`. Also the method is used in two other cases with a constant string directly. Therefore, I did not want to change it as in those two cases it doesn’t really make sense to wrap the error specifically.

Wrapping the string inside an Error isn’t really necessary in this pipe-block either, but I prefer it here now after the change as having the error-subscription using an actual error makes sense, I think.

Reply:
Fine